### ISSUE TITLE
[PyTorch][Feature Request]Lookahead Optimizer

### ISSUE THREAD

**Issue by PistonY** [pytorch/pytorch#23366]
## 🚀 Lookahead Optimizer
https://arxiv.org/pdf/1907.08610.pdf

In this paper they show new Optimizer can save time, improve accuracy than SGD and work both on CV/NLP.
I think it may worth to try.

**Comment by illcat** [comment]
I think so.

**Comment by travishsu** [comment]
It somehow looks like Stochastic Weight Averaging, could be plugged into every existed optimizer. Maybe it's more appropriate to write some wrapper class like scheduler.

**Comment by PistonY** [comment]
@travishsu I noticed that too, and it seems not update model parameters every batch/iter but find a good status then update. Looks like different between existed optimizer.

**Comment by soumith** [comment]
We don't think this method should be in pytorch core, as opposed to your own personal repository or something like https://github.com/pytorch/contrib , at least not yet in time.

Our reservation is that we want to include methods that the community uses as a standard, or else the code maintenance problem balloons up for us.
We do show discretion based on what the paper shows as evidence, for example BatchNorm was included within weeks of it's publish date (in Torch).
In terms of rejected methods, we've rejected newly minted papers such as Swish ( #3260 , #3182 ), Yellowfin ( https://github.com/pytorch/pytorch/issues/1960 ) and many others, and rightly so, these haven't become standardized in the community (like LSTM / Transformer / BatchNorm).

If you have a differing opinion, let us know why, and we can re-think.

tl;dr: The paper doesn't show evidence that makes it a method that has obvious long-term success. If the paper does have long-term success in the field we will include it

**Comment by PistonY** [comment]
I understood, just a suggestion. I don't think it would become a standardized method when I finish reading the paper in detail, hoping I can try it sometimes.
